### PR TITLE
better tests

### PR DESCRIPTION
**Pull request by eishay**

### CODE REVIEW

**Comment by eishay** [comment]
@andrewconner @ahsu1230 @raymondng see new api to create and connect users

``` scala
+          val userGW = user().withName("George", "Washington").withUsername("GDubs").saved
-          val userAL = userRepo.save(User(firstName = "Abe", lastName = "Lincoln", username = Username("abe"), normalizedUsername = "abe"))    
+          val userAL = user().withName("Abe", "Lincoln").withUsername("abe").saved.savedConnection(userGW)
-          val userTJ = userRepo.save(User(firstName = "Thomas", lastName = "Jefferson", username = Username("TJ"), normalizedUsername = "tj"))   
+          val userTJ = user().withName("Thomas", "Jefferson").withUsername("TJ").saved
-          val userJA = userRepo.save(User(firstName = "John", lastName = "Adams", username = Username("jayjayadams"), normalizedUsername = "jayjayadams"))   
+          val userJA = user().withName("John", "Adams").withUsername("jayjayadams").saved
-          val userBF = userRepo.save(User(firstName = "Ben", lastName = "Franklin", username = Username("Benji"), normalizedUsername = "benji"))   
+          val userBF = user().withName("Ben", "Franklin").withUsername("Benji").saved
-          userConnectionRepo.save(UserConnection(user1 = userGW.id.get, user2 = userAL.id.get))    
-          userConnectionRepo.save(UserConnection(user1 = userGW.id.get, user2 = userTJ.id.get))    
-          userConnectionRepo.save(UserConnection(user1 = userJA.id.get, user2 = userGW.id.get))    
+          connect(userGW -> userAL, userGW -> userTJ, userJA -> userGW).saved
```

https://github.com/eishay/keepit/compare/user_profile_header?expand=1#diff-dba533fe2f4197750ae253134d9d613aR302

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserControllerTest.scala [hunk 1]
```diff
@@ -289,27 +293,46 @@ class UserControllerTest extends Specification with ShoeboxTestInjector {
...
-          val userAL = userRepo.save(User(firstName = "Abe", lastName = "Lincoln", username = Username("abe"), normalizedUsername = "abe"))
-          val userTJ = userRepo.save(User(firstName = "Thomas", lastName = "Jefferson", username = Username("TJ"), normalizedUsername = "tj"))
-          val userJA = userRepo.save(User(firstName = "John", lastName = "Adams", username = Username("jayjayadams"), normalizedUsername = "jayjayadams"))
-          val userBF = userRepo.save(User(firstName = "Ben", lastName = "Franklin", username = Username("Benji"), normalizedUsername = "benji"))
-          userConnectionRepo.save(UserConnection(user1 = userGW.id.get, user2 = userAL.id.get))
-          userConnectionRepo.save(UserConnection(user1 = userGW.id.get, user2 = userTJ.id.get))
-          userConnectionRepo.save(UserConnection(user1 = userJA.id.get, user2 = userGW.id.get))
+          val userGW = user().withName("George", "Washington").withUsername("GDubs").withPictureName("pic1").saved
+          val userAL = user().withName("Abe", "Lincoln").withUsername("abe").saved.savedConnection(userGW)
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserControllerTest.scala:297]: oh nice!

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserControllerTest.scala [hunk 2]
```diff
@@ -289,27 +293,46 @@ class UserControllerTest extends Specification with ShoeboxTestInjector {
...
+          val userTJ = user().withName("Thomas", "Jefferson").withUsername("TJ").saved
+          val userJA = user().withName("John", "Adams").withUsername("jayjayadams").saved
+          val userBF = user().withName("Ben", "Franklin").withUsername("Benji").saved
+
+          connect(userGW -> userAL,
+            userGW -> userTJ,
+            userJA -> userGW,
+            userAL -> userTJ).saved
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserControllerTest.scala:305]: even cooler

**Comment by ahsu1230** [comment]
:+1:

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserControllerTest.scala [hunk 3]
```diff
@@ -289,27 +293,46 @@ class UserControllerTest extends Specification with ShoeboxTestInjector {
...
         res === Json.parse(
           """
-            |{"firstName":"George","lastName":"Washington","pictureName":null,"numLibraries":0,"numKeeps":0,"numFriends":3,"numFollowers":0,"helpedRekeep":0}
+            |{
+            |  "firstName":"George",
+            |  "lastName":"Washington",
+            |  "pictureName":"pic1",
+            |  "numLibraries":0
+            |}
           """.stripMargin)
```
Comment (2is10) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserControllerTest.scala:336]: Why do so many of our tests use `.stripMargin` like this? The `JSON.parse` call alone would eliminate the extra whitespace.

**Comment by 2is10** [comment]
nice work!

File: kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserControllerTest.scala (hunk 3 above)
Comment (eishay) [inline kifi-backend/modules/shoebox/test/com/keepit/controllers/website/UserControllerTest.scala:336]: i'll remove it, no need for that. thansk

**Comment by raymondng** [comment]
:+1: The less code involved in scaffolding, the better. This also makes the intent clear. Thanks @eishay!

**Pull request merged**